File: lib_basissdk/src/main/java/com/cjh/lib_basissdk/util/CloseUtils.java
package com.cjh.lib_basissdk.util;

import java.io.Closeable;
import java.io.IOException;

/**
 * 关闭相关工具类 
 */
public final class CloseUtils {
	
	private CloseUtils() {
        throw new UnsupportedOperationException("u can't instantiate me...");
    }
	
	/**
     * 关闭 IO
     *
     * @param closeables closeables
     */
	public static void closeIO(final Closeable... closeables) {
		if (null == closeables) return;
		for (Closeable closeable : closeables) {
            if (closeable != null) {
                try {
                    closeable.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
	}
	
	/**
     * 安静关闭 IO(不打印IO异常就是安静？)
     *
     * @param closeables closeables
     */
    public static void closeIOQuietly(final Closeable... closeables) {
        if (closeables == null) return;
        for (Closeable closeable : closeables) {
            if (closeable != null) {
                try {
                    closeable.close();
                } catch (IOException ignored) {
                }
            }
        }
    }
	
}

File: lib_basissdk/src/main/java/com/cjh/lib_basissdk/util/CloseUtils.kt
package com.cjh.lib_basissdk.util

import java.io.Closeable
import java.io.IOException

/**
 * @author: caijianhui
 * @date: 2020/4/29 17:16
 * @description: 关闭相关工具类
 */
object CloseUtils {

    /**
     * 关闭 IO
     *
     * @param closeables closeables
     */
    fun closeIO(vararg closeables: Closeable?) {
        if (null == closeables) return
        for (closeable in closeables) {
            if (closeable != null) {
                try {
                    closeable.close()
                } catch (e: IOException) {
                    e.printStackTrace()
                }
            }
        }
    }


    /**
     * 安静关闭 IO(不打印IO异常就是安静？)
     *
     * @param closeables closeables
     */
    fun closeIOQuietly(vararg closeables: Closeable?) {
        if (closeables == null) return
        for (closeable in closeables) {
            if (closeable != null) {
                try {
                    closeable.close()
                } catch (ignored: IOException) {
                }
            }
        }
    }


}